client-config/build.gradle

```diff
@@ -5,6 +5,8 @@ apply plugin: 'com.palantir.revapi'
 dependencies {
     api 'com.palantir.conjure.java.api:service-config'
     api 'com.palantir.tritium:tritium-registry'
+    api 'com.google.errorprone:error_prone_annotations'
+    api 'com.palantir.refreshable:refreshable'
```

Comment:
I could move this stuff to a separate jar if we want? like `com.palantir.conjure.java.runtime:client-factories`?

Reply:
client-factory-api maybe?

Reply:
I don't think we really buy anything from splitting them though, we're forced into the api dependency on ClientCofiguration.<enum>. If we are interested in breaking that, we should put the client factory api in cjr-api instead.

Reply:
What if we didn't depend on the ClientConfiguration enums and allowed those via mixins elsewhere?

Reply:
I mean it's feasible to put n-1 interfaces in c-j-r-api but leave the enum-related mixin here, just seems like the sprawl might end up a bit of a pain.